content/local-setup.md

```diff
@@ -0,0 +1,283 @@
+- `Keypair Path` - the keypair used when running CLI commands (`id.json` is created by default)
+
+To change the `RPC URL`, use the `solana config set --url` command.
+
+```bash
+solana config set --url http://localhost:8899
+```
+
+```bash
+solana config set --url https://api.devnet.solana.com
+```
+
+```bash
+solana config set --url https://api.mainnet-beta.solana.com
+```
+
+You can generate a new keypair using the `solana-keygen new --outfile` command followed by the file path to store the keypair.
+
+```bash
+solana-keygen new --outfile ~/<FILE_PATH>
+```
+
+To update the `Keypair Path` for the Solana CLI config, using the `solana config set --keypair` command and specify the path of the new keypair (Solana CLI will now use the keypair from the specified path when running commands).
+
+```bash
+solana config set --keypair ~/<FILE_PATH>
+```
+
+To view the `publickey` of the currently configured keypair, run the `solana address` command.
+
+```bash
+solana address
+```
+
+To view the SOL balance of the of the currently configured keypair, run the `solana balance` command.
+
+```bash
+solana balance
+```
+
+To airdrop SOL for testing (on Devnet or Localhost), run the `solana airdrop <AMOUNT>` command (Note that while on Devnet you are limited to 2 SOL per airdrop).
+
+```bash
+solana airdrop 2
+```
+
+While deploying and testing on Localhost, you must run a test validator using the `solana-test-validator` command.
+
+```bash
+solana-test-validator
+```
+
+While on Localhost, you can view program logs by running the `solana logs` command in a separate terminal.
+
+```bash
+solana logs
+```
+
+To view program logs while on Devnet, use the `solana logs | grep "<PROGRAM_ID> invoke"` command and specify the program Id. If the program Id is not specified, then the `solana logs` command will return a constant stream of logs from Devnet. You can then specify the number of lines (of program logs) to return by including `-A <NUMBER_OF_LINES_TO_RETURN>` after the command.
+
+```bash
+solana logs | grep "<PROGRAM_ID> invoke" -A <NUMBER_OF_LINES_TO_RETURN>
+```
+
+To deploy a program, you using the `solana program deploy` command and specify the file path of the program. The command to deploy a program along with the file path is provided as an output when you build the program. We’ll go over this shortly in the demo.
+
+```bash
+solana program deploy <PATH>
+```
+
+Lastly, you can view the details for Solana CLI commands using the `solana --help` command.
+
+```bash
+solana --help
+```
+
+## Solana Program from Local Environment
+
+To create a new Rust project run the following in a new terminal.
+
+```bash
+cargo new --lib <PROJECT_FILE_NAME>
+```
+
+Open the new project and update the `Cargo.toml` file. Remember to specify the latest version of `solana-program`.
+
+```rust
+[package]
+name = "<PROJECT_FILE_NAME>"
+version = "0.1.0"
+edition = "2021"
+
+[features]
+no-entrypoint = []
+
+[dependencies]
+solana-program = "1.10.7"
```

Comment:
@ZYJLiu sorry, dumb question - should the latest `solana-program` crate's `"version #"` match the return from the `% solana --version` command?

Asking because above we have them install a later version (1.10.24) - code on line 55

Also, asking cuz my CLI install tells me weird things:

```sh
% solana program --version
solana-program
```
🤷🏻‍♂️

Currently Solana CLI is at 1.10.26.

Should we mention a best practice (?) of keeping up to date, or noting the update command after the installation? E.g. `solana-install update` - available on the linked page in the "Download Solana CLI" section

Reply:
Same question re: the Rust code block below, specifically line 229

Reply:
should we just link to official pages for download instead of providing commands within the tutorial?

the rust dependencies code block was more to demo what the file should look like. Should I change to something more generic like `solana-program = <VERSION>`